staging/src/k8s.io/csi-translation-lib/plugins/vsphere_volume.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// VSphereDriverName is the name of the CSI driver for vSphere Volume
+	VSphereDriverName = "csi.vsphere.vmware.com"
+	// VSphereInTreePluginName is the name of the in-tree plugin for vSphere Volume
+	VSphereInTreePluginName = "kubernetes.io/vsphere-volume"
+
+	// paramStoragePolicyName used to supply SPBM Policy name for Volume provisioning
+	paramStoragePolicyName = "storagepolicyname"
+
+	// This param is used to tell Driver to return volumePath and not VolumeID
+	// in-tree vSphere plugin does not understand volume id, it uses volumePath
+	paramcsiMigration = "csimigration"
+
+	// This param is used to supply datastore name for Volume provisioning
+	paramDatastore = "datastore"
```

Comment:
Since these translate to CSI storage class params, can we use a different value for migration params like `datastoreMigrationParam`, `diskformatMigrationParam` ... ? Basically suffix  VCP param with `MigrationParam`.

Reply:
> can we use a different value for migration params like datastoreMigrationParam

suffixed legacy VCP params with `-migrationparam`  - 

Please see - https://github.com/kubernetes/kubernetes/pull/90911/commits/4193e6f0a540bf918453db8b870ffd66862d3241

> can you also file an issue in vsphere CSI repo to document caveats of these params, why it should not be used in CSI storage class, etc?

Issue: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/236